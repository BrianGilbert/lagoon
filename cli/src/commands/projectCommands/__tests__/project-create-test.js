// @flow

import R from 'ramda';
import { runGQLQuery } from '../../../query';

import {
  allOptionsSpecified,
  commandOptions,
  getAllowedCustomersAndOpenshifts,
} from '../create';

jest.mock('../../../query');

const _mock = (mockFn: any): JestMockFn => mockFn;

describe('allOptionsSpecified', () => {
  it('should return true when all options are specified', () => {
    const returnVal = allOptionsSpecified(commandOptions);
    expect(returnVal).toBe(true);
  });

  it('should return true when all options and more are specified', () => {
    const returnVal = allOptionsSpecified({
      ...commandOptions,
      anotherOptionKey: 'anotherOptionValue',
    });
    expect(returnVal).toBe(true);
  });

  it('should return false if not all options are specified', () => {
    const returnVal = allOptionsSpecified({
      ...R.dissoc(R.head(R.keys(commandOptions)), commandOptions),
    });
    expect(returnVal).toBe(false);
  });
});

describe('getAllowedCustomersAndOpenshifts', () => {
  it('should return all customers and all openshifts', async () => {
    _mock(runGQLQuery).mockImplementationOnce(() =>
      Promise.resolve({
        data: {
          allCustomers: [
            {
              id: 1,
              name: 'credentialtest-customer1',
              comment: 'used to test the cli',
              private_key: null,
              created: '2018-02-20 19:38:11',
            },
            {
              id: 2,
              name: 'credentialtest-customer2',
              comment: 'used to test the cli',
              private_key: null,
              created: '2018-02-20 19:38:11',
            },
          ],
          allOpenshifts: [
            {
              id: 1,
              name: 'credentialtest-openshift',
              console_url: 'https://localhost:8443/',
              token: null,
              router_pattern: null,
              project_user: null,
              ssh_host: null,
              ssh_port: null,
              created: '2018-02-20 19:38:11',
            },
          ],
        },
      }));

    const cerr = jest.fn();

    const returnVal = await getAllowedCustomersAndOpenshifts(cerr);

    expect(R.prop('allCustomers', returnVal)).toHaveLength(2);
    expect(R.prop('allOpenshifts', returnVal)).toHaveLength(1);
    expect(returnVal).toMatchSnapshot();
  });
});
