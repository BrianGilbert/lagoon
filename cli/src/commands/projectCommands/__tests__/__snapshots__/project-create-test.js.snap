// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAllowedCustomersAndOpenshifts should return all customers and all openshifts 1`] = `
Object {
  "allCustomers": Array [
    Object {
      "comment": "used to test the cli",
      "created": "2018-02-20 19:38:11",
      "id": 1,
      "name": "credentialtest-customer1",
      "private_key": null,
    },
    Object {
      "comment": "used to test the cli",
      "created": "2018-02-20 19:38:11",
      "id": 2,
      "name": "credentialtest-customer2",
      "private_key": null,
    },
  ],
  "allOpenshifts": Array [
    Object {
      "console_url": "https://localhost:8443/",
      "created": "2018-02-20 19:38:11",
      "id": 1,
      "name": "credentialtest-openshift",
      "project_user": null,
      "router_pattern": null,
      "ssh_host": null,
      "ssh_port": null,
      "token": null,
    },
  ],
  "errors": undefined,
}
`;
